config.h

```diff
@@ -0,0 +1,216 @@
+#pragma once
+
+#include "lib/types.h"
+
+#include <docopt/docopt.h>
+
+#include <map>
+#include <sstream>
+
+/**
+ * Common configuration
+ */
+struct Config {
+    // image
+    float aspect = 1;
+    size_t width = 640;
+
+    // common options
+    size_t num_threads = 1;
+    float inverse_gamma = 0.454545;
+    float exposure = 1;
+    Color bg_color;
+    bool gamma_correction_enabled = true;
+
+    // scene
+    std::string filename;
+
+    void check() const {
+        assert(0 < aspect);
+        assert(1 <= num_threads);
+        assert(0 <= exposure);
+    }
+
+    /**
+     * Parse color from string.
+     *
+     * @param  color_str contains either
+     *         1. one numerical value in [0, 255], or
+     *         2. three space-separated numerical values in [0, 255].
+     * @return if color_str is of type 1, then return Color with constant value
+     *         for all three channels, otherwise Color with channels containing
+     *         corresponding parsed values.
```

Comment:
Actually it is a todo. We should definitely use 3-channel color everywhere, unless we need 4-channel explicitly.